### PR TITLE
Feat: add file download api with local telegram api

### PR DESCRIPTION
**Pull request by icceey**
This pull request introduces several significant changes, including new GitHub workflows, Dockerfile setup, and the main application logic for a Telegram bot API. The changes are primarily focused on setting up CI/CD pipelines, Docker image creation, and the core application functionality using Rust and Axum.

### GitHub Workflows and CI/CD:
* Added `.github/workflows/build.yml` to define a build process using GitHub Actions, which includes steps for checking out the source code, building the project, and running tests.
* Added `.github/workflows/pull-request.yml` to trigger the build workflow on pull requests and build a Docker image.
* Added `.github/workflows/push.yml` to trigger the build and deploy workflow on push events, excluding dependabot branches.
* Added `.github/workflows/release.yml` to build and push Docker images to GitHub Container Registry on release events.

### Dockerfile Setup:
* Created a `Dockerfile` to build the Telegram bot API and the Rust application, including multi-stage builds for optimized image size.

### Application Logic:
* Implemented the main application logic in `src/main.rs` using Axum for routing and handling HTTP requests, including proxying requests to the Telegram bot API and serving files.

### Configuration and Dependencies:
* Added `Cargo.toml` to define the Rust project configuration and dependencies, including Axum, Clap, Reqwest, and Tokio.

### Additional Configuration:
* Added `.dockerignore` to exclude unnecessary files from the Docker build context.
* Added `.github/dependabot.yml` to configure Dependabot for managing Cargo dependencies with daily updates.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: 2
+updates:
+  - package-ecosystem: cargo
+    directory: "/"
+    schedule:
+      interval: daily
+      time: "19:00"
+    open-pull-requests-limit: 10
+    target-branch: dev
```
Comment (icceey) [inline .github/dependabot.yml:9]: should be master

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+jobs:
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get source
+        uses: actions/checkout@v4
+      - name: Build
+        run: cargo build --verbose
```
Comment (hugefiver) [inline .github/workflows/build.yml:18]: 感觉可以删除 cargo build

Comment (hugefiver) [reply .github/workflows/build.yml:18]: 使用 cargo check 可以大概

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
+FROM alpine AS builder
+
+RUN apk add --no-cache cmake openssl-dev gcc g++ gperf zlib-dev git make linux-headers musl
+
+RUN git clone --depth 1 --recursive https://github.com/tdlib/telegram-bot-api.git
+RUN CC=musl-cc CXX=musl-c++ cd telegram-bot-api && mkdir build && cd build && \
+    cmake -DCMAKE_BUILD_TYPE=Debug .. && make -j 8
```
Comment (hugefiver) [inline Dockerfile:7]: 使用 `make -j $(nproc)`

**Review by hugefiver — COMMENTED** [review]
reviewed

**Review by hugefiver — DISMISSED** [review]

**Review by hugefiver — APPROVED** [review]

**Pull request merged**